modules/infra/control/control_output.c

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include <gr_control.h>
+#include <gr_control_output.h>
+#include <gr_graph.h>
+#include <gr_log.h>
+#include <gr_macro.h>
+#include <gr_mbuf.h>
+
+#include <event2/event.h>
+#include <rte_ether.h>
+#include <rte_graph_worker.h>
+
+#include <pthread.h>
+#include <stdatomic.h>
+#include <stdlib.h>
+
+extern struct rte_ring *control_output_ring;
+static struct event *control_output_evt;
+
+static int tshutdown;
+pthread_t thread_id;
+pthread_cond_t cv;
+pthread_mutex_t mp;
+
+static void poll_control_output_ring(evutil_socket_t, short, void *) {
+	struct gr_control_output_msg msg[RTE_GRAPH_BURST_SIZE];
+	int n = 0;
+
+	while ((n = rte_ring_dequeue_burst_elem(
+			control_output_ring, msg, sizeof(msg[0]), ARRAY_DIM(msg), NULL
+		))
+	       > 0) {
+		for (int i = 0; i < n; i++) {
+			msg[i].callback(msg[i].mbuf);
```

Comment:
Just an open question.
Do we want to batch those callback calls?

Reply:
this should be only running on the control plane thread, and I don't expect (yet) a huge amount of traffic between the 2.
I suggest to revisit that when we'll hit a performance issue.

Reply:
Sure.